tests/build.rs

```diff
@@ -77,81 +37,97 @@ fn clean_target_binaries<P: AsRef<Path>>(path: P) -> anyhow::Result<()> {
     Ok(())
 }
 
-fn ensure_target_exists<P: AsRef<Path>>(path: P) -> anyhow::Result<()> {
-    let artifact_path = target_path().join(path);
+fn ensure_path_exists<P: AsRef<Path>>(path: P) -> anyhow::Result<()> {
     anyhow::ensure!(
-        fs::exists(&artifact_path)?,
+        fs::exists(path.as_ref())?,
         "artifact {} does not exist",
-        artifact_path.display()
+        path.as_ref().display()
     );
     Ok(())
 }
 
-/// The bevy CLI command, executed in the test repository.
-fn bevy() -> Command {
-    let mut cmd = Command::new("bevy");
-    cmd.current_dir(test_path());
-    cmd
+fn executable(binary: &str) -> String {
+    if cfg!(windows) {
+        format!("{binary}.exe")
+    } else {
+        binary.to_string()
+    }
 }
 
 #[test]
 #[serial]
-fn should_build_native_dev() -> anyhow::Result<()> {
-    setup()?;
-    let target_path = Path::new("debug");
-    clean_target_binaries(target_path)?;
-    bevy()
-        .args(["build", "-p=bevy_default", "--yes"])
-        .ensure_status()?;
-
-    ensure_target_exists(target_path.join(executable("bevy_default")))
+fn should_build_native_debug() -> anyhow::Result<()> {
+    let mut cmd = Command::cargo_bin("bevy")?;
+    cmd.current_dir(test_path())
+        .args(["build", "-p=bevy_default", "--yes"]);
+
+    let target_artifact_path = target_path().join("debug");
+
+    clean_target_artifacts(&target_artifact_path)?;
+
+    cmd.assert().success();
+
+    ensure_path_exists(target_artifact_path.join(executable("bevy_default")))
         .context("binary does not exist")
 }
 
 #[test]
 #[serial]
 fn should_build_native_release() -> anyhow::Result<()> {
-    setup()?;
-    let target_path = Path::new("release");
-    clean_target_binaries(target_path)?;
-    bevy()
-        .args(["build", "-p=bevy_default", "--yes", "--release"])
-        .ensure_status()?;
-
-    ensure_target_exists(target_path.join(executable("bevy_default")))
+    let mut cmd = Command::cargo_bin("bevy")?;
+    cmd.current_dir(test_path())
+        .args(["build", "-p=bevy_default", "--yes", "--release"]);
+
+    let target_artifact_path = target_path().join("release");
+
+    clean_target_artifacts(&target_artifact_path)?;
```

Comment:
Can we put these at the start of the test, just to make it clear that we run this before the `bevy` execution?